Fix part of #10049: Remove deprecated fields from `CollectionModel` and `CollectionRightsModel`

iramin opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #10049 .
2. This PR does the following: Remove deprecated fields from collection models: node from CollectionModel  and translator_ids  from CollectionRightsModel.

## Essential Checklist

- [X] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [X] The linter/Karma presubmit checks have passed locally on your machine.
- [X] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [X] The PR is made from a branch that's **not** called "develop".

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#instructions-for-making-a-code-change).
- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.

iramin committed:
Remove deprecated translator_ids from CollectionRightsModel

iramin committed:
Remove deprecated translator_ids from CollectionRightsModel

iramin committed:
Remove deprecated field 'nodes' from CollectionModel

iramin committed:
Remove deprecated field 'nodes' from CollectionModel

iramin committed:
Merge remote-tracking branch 'oppia_origin/develop' into remove-deprecated-collection-models

seanlip commented:
@vojtechjelinek, just to check -- should you be first reviewer on this PR?

Also, @iramin, please note the failing backend and Karma tests. Thanks!

vojtechjelinek commented on core/domain/collection_jobs_one_off.py:
```diff
@@ -96,3 +96,82 @@ def reduce(key, values):
+            del collection_rights_model._properties['translator_ids']  # pylint: disable=protected-access
+            if 'translator_ids' in collection_rights_model._values:  # pylint: disable=protected-access
+                del collection_rights_model._values['translator_ids']  # pylint: disable=protected-access
+            collection_rights_model.put(update_last_updated_time=False)
```
@iramin This won't work anymore, use `collection_rights_model.update_timestamps(update_last_updated_time=False)` and then `collection_rights_model.put()` ditto below

vojtechjelinek replied:
This is probably the reason why the backend tests fail

vojtechjelinek commented on core/storage/collection/gae_models.py:
```diff
@@ -108,15 +104,69 @@ def get_export_policy(cls):
+        # need to move the values from nodes field to collection_contents dict
+        # and delete nodes.
+        if 'nodes' in model_dict and model_dict['nodes']:
+            if ('collection_contents' not in model_dict or not
+                    model_dict['collection_contents']
+               ):
+                model_dict['collection_contents'] = (
+                    (datastore_services.JsonProperty(
+                        default={}, indexed=False)))
```
This part is not needed, the `collection_contents` will always exist as long as it is defined above (line 82).

vojtechjelinek commented on core/storage/collection/gae_models.py:
```diff
@@ -108,15 +104,69 @@ def get_export_policy(cls):
+                        default={}, indexed=False)))
+            model_dict['collection_contents']['nodes'] = (
+                copy.deepcopy(model_dict['nodes']))
+            model_dict['nodes'] = []
```
```suggestion
            del model_dict['nodes']
```

vojtechjelinek commented on core/domain/collection_jobs_one_off.py:
```diff
@@ -96,3 +96,82 @@ def reduce(key, values):
+        # see https://stackoverflow.com/a/15116016/3688189 and
+        # https://stackoverflow.com/a/12701172/3688189.
+        if 'translator_ids' in collection_rights_model._properties:  # pylint: disable=protected-access
+            del collection_rights_model._properties['translator_ids']  # pylint: disable=protected-access
```
Since now we delete it in the `convert_to_valid_dict` we should do `del model_dict['translator_ids']` instead of `model_dict['translator_ids'] = []`

vojtechjelinek reviewed (changes requested):
Thanks! Left a few comments.

iramin committed:
Merge remote-tracking branch 'oppia_origin/develop' into remove-deprecated-collection-models

vojtechjelinek commented:
@iramin Hi, do you need help with anything on this PR?

iramin commented:
> 
> 
> @iramin Hi, do you need help with anything on this PR?

I followed the suggested steps, but I get the following error in the backend tests:
https://github.com/oppia/oppia/pull/11350/checks?check_run_id=1517506339#step:8:358

Can you help with this? What have I missed?

vojtechjelinek commented:
@iramin I have left some comments in the code (see https://github.com/oppia/oppia/pull/11350#discussion_r539416830) these should help with the errors.

iramin commented:
> 
> 
> @iramin I have left some comments in the code (see [#11350 (comment)](https://github.com/oppia/oppia/pull/11350#discussion_r539416830)) these should help with the errors.

yeah, I applied those comments. still, I get the error about 'put'. Should I use put or commit for the versioned model?

vojtechjelinek commented:
@iramin Oh, yes, you should use `commit`

iramin committed:
Fix one_off job test errors for collection models

Deepam101 committed:
Migration (topic-editor-routing.service.ts) AngularJS to Angular 8 (#11188)

* Migrate tpic-routing

* Added the activeTabs

* Added the activeTabs

* Fixed tests

* Fixed tests

* Fixed window tests

* changes

* Fixed tests

* linting changes

* More linting changes

* More linting changes

* More linting changes

* Fixed TopicEditorRoutingService linting

* added missing test

* Added more missing test

* Changes in URL navigation

* Fixed URL for tests and server

* Linting changes

* Fixed faulty test

* requested changes

* Comment changes

* Requested changes

Co-authored-by: Srijan Reddy <srijanreddy98@gmail.com>
Co-authored-by: Deepam101 <deepambanerjee07@gmail.com>

srijanreddy98 committed:
Revert "Migration (topic-editor-routing.service.ts) AngularJS to Angular 8 (#11188)" (#11423)

This reverts commit 193e8d674b03522a5d1d1df8466f43ea5fef05d3.

vojtechjelinek commented:
@iramin The tests are still failing it is due to https://github.com/oppia/oppia/blob/develop/core/domain/collection_services.py#L117-L121 which can be removed now that we move this functionality to reconstitute. Als, when you want another review from me please assign me.

seanlip committed:
Revert "Fix #11140: Remove retrieval of `UserAuthDetailsModel` (#11363)" (#11443)

This reverts commit 6c8b3fde1e7126f1bb980a20d4e930b47d8f827c.

aks681 committed:
Modify the splash page (#11260)

* 4 sections done

* minor change

* Desktop page done

* lint

* fixed lint and TS errors

* minor change

* added mobile view

* Addressed review comments

* fixed backend test

* fixed frontend test

* Addressed comments

* Addressed more comments

* review comment

* addressed review comments and minor UI changes

* Addressed review comments

iramin committed:
Merge remote-tracking branch 'oppia_origin/develop' into remove-deprecated-collection-models

iramin committed:
Fix collection one off job errors

vojtechjelinek commented on core/domain/collection_jobs_one_off_test.py:
```diff
@@ -21,16 +21,22 @@
+
+datastore_services = models.Registry.import_datastore_services()
+
+datastore_services = models.Registry.import_datastore_services()
```
Duplication

iramin commented:
> 
> 
> @iramin The tests are still failing it is due to https://github.com/oppia/oppia/blob/develop/core/domain/collection_services.py#L117-L121 which can be removed now that we move this functionality to reconstitute. Als, when you want another review from me please assign me.

@vojtechjelinek I have an [error](https://github.com/oppia/oppia/pull/11350/checks?check_run_id=1583658779#step:8:6057) in this function: https://github.com/oppia/oppia/blob/4c32e0440bbac7f775f11445e31053ddeb66c7db/core/domain/collection_jobs_one_off_test.py#L182
Is this still needed?
Also, there are some other issues in functions like this:

https://github.com/oppia/oppia/blob/4c32e0440bbac7f775f11445e31053ddeb66c7db/core/domain/collection_domain.py#L470 

I couldn't figure out if this is needed or not.

vojtechjelinek reviewed (changes requested):
@iramin Probably this https://github.com/oppia/oppia/blob/4c32e0440bbac7f775f11445e31053ddeb66c7db/core/domain/collection_domain.py#L479 and this code will need to be removed https://github.com/oppia/oppia/blob/4c32e0440bbac7f775f11445e31053ddeb66c7db/core/domain/collection_domain.py#L509

The tests are probably failing because of this stuff above

iramin committed:
Merge remote-tracking branch 'oppia_origin/develop' into remove-deprecated-collection-models

iramin committed:
Fix collection one off job errors

iramin commented:
@vojtechjelinek This is where the one_off tests are failing:
https://github.com/oppia/oppia/pull/11350/checks?check_run_id=1596408358#step:8:3567

vojtechjelinek commented on core/domain/collection_jobs_one_off_test.py:
```diff
@@ -133,143 +137,466 @@ def test_migrate_collections_failing_strict_validation(self):
+        # Setup user who will own the test objects.
+        self.signup(self.ALBERT_EMAIL, self.ALBERT_NAME)
+        self.albert_id = self.get_user_id_from_email(self.ALBERT_EMAIL)
+        self.process_and_flush_pending_mapreduce_tasks()
```
@iramin I would try removing this, I have a slight suspicion that it leads to the errors.

iramin replied:
@vojtechjelinek 
It is used in other tests that don't fail. for example:
https://github.com/oppia/oppia/blob/c592384dd2327b8fa1aa6d00a4527a32b5609062/core/domain/collection_jobs_one_off_test.py#L52

Other than that, it seems that I need a user to commit the model at:
https://github.com/oppia/oppia/blob/c592384dd2327b8fa1aa6d00a4527a32b5609062/core/domain/collection_jobs_one_off_test.py#L547

vojtechjelinek replied:
Hmmm, still worth testing it though? If it won't work I will take a look tomorrow when I'm back home to my main PC.

iramin replied:
Just tried it and got the test failed again.

iramin committed:
Merge remote-tracking branch 'oppia_origin/develop' into remove-deprecated-collection-models

vojtechjelinek commented on core/domain/collection_jobs_one_off_test.py:
```diff
@@ -133,143 +137,466 @@ def test_migrate_collections_failing_strict_validation(self):
+        (
+            collection_jobs_one_off
+            .RemoveCollectionRightsTranslatorIdsOneOffJob.enqueue(job_id))
+        self.assertEqual(
+            self.count_jobs_in_taskqueue(
+                taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
+        self.process_and_flush_pending_tasks()
```
```suggestion
        self.assertEqual(
            self.count_jobs_in_mapreduce_taskqueue(
                taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
        self.process_and_flush_pending_mapreduce_tasks()
```

vojtechjelinek commented on core/domain/collection_jobs_one_off_test.py:
```diff
@@ -133,143 +137,466 @@ def test_migrate_collections_failing_strict_validation(self):
+        (
+            collection_jobs_one_off
+            .RemoveCollectionModelNodesOneOffJob.enqueue(job_id))
+        self.assertEqual(
+            self.count_jobs_in_taskqueue(
+                taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
+        self.process_and_flush_pending_tasks()
```
```suggestion
        self.assertEqual(
            self.count_jobs_in_mapreduce_taskqueue(
                taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
        self.process_and_flush_pending_mapreduce_tasks()
```

vojtechjelinek reviewed (changes requested):
Here is how to fix the tests, there are other errors in the tests that you will need to look into.

DubeySandeep commented:
@iramin, Any update on this PR? Do let me know if you need any help?

iramin committed:
Merge remote-tracking branch 'oppia_origin/develop' into remove-deprecated-collection-models

iramin committed:
fix errors in remove collection deprecated fields

iramin committed:
--amend

iramin commented:
> 
> 
> @iramin, Any update on this PR? Do let me know if you need any help?

@DubeySandeep 
It seems that the tests are failing in the one_off tests, because the changes do not take effect:
https://github.com/oppia/oppia/runs/1637129937?check_suite_focus=true#step:8:5273

Is it the correct way of updating a versioned model?

vojtechjelinek commented:
@iramin Please check an email thread with Michael Wu where I added you, I mention some stuff you will need to do in order to properly test this. You can also check #11529 especially the mock class in the tests.

iramin committed:
Merge remote-tracking branch 'oppia_origin/develop' into remove-deprecated-collection-models

iramin committed:
fix one_off_job test failures in remove collection deprecated fields

iramin commented:
> 
> 
> @iramin Please check an email thread with Michael Wu where I added you, I mention some stuff you will need to do in order to properly test this. You can also check #11529 especially the mock class in the tests.

@vojtechjelinek Thanks! I did not receive the email thread that you mentioned, but, by looking at #11529 I fixed the way that I commit collections. The tests now pass on my local machine. Please take a look and let me know if you have any comments.

vojtechjelinek commented on core/domain/collection_services.py:
```diff
@@ -672,6 +672,10 @@ def apply_change_list(collection_id, change_list):
+            elif (change.cmd ==
+                  collection_domain.CMD_REMOVE_NODE):
+                # No action is needed here.
+                continue
```
Why is this needed? And also probably change it to
```suggestion
            elif (change.cmd == collection_domain.CMD_REMOVE_NODE):
                # No action is needed here.
                continue
```

iramin replied:
removed

vojtechjelinek commented on core/domain/collection_domain.py:
```diff
@@ -503,10 +508,6 @@ def _convert_v5_dict_to_v6_dict(cls, collection_dict):
         This changes the structure of each node to not include skills as well
         as remove skills from the Collection model itself.
         """
-        new_collection_dict = (
-            cls._convert_collection_contents_v5_dict_to_v6_dict(
-                collection_dict))
```
Why is this part removed too?

iramin replied:
Because the variable 'new_collection_dict' is unused now.

vojtechjelinek replied:
Ah, I see, thanks.

vojtechjelinek commented on core/domain/collection_domain.py:
```diff
@@ -144,6 +145,11 @@ class CollectionChange(change_domain.BaseChange):
+        'name': CMD_REMOVE_NODE,
+        'required_attribute_names': [],
+        'optional_attribute_names': [],
+        'user_id_attribute_names': []
```
Why is this added?

iramin replied:
removed

vojtechjelinek commented on core/domain/collection_jobs_one_off.py:
```diff
@@ -96,3 +96,84 @@ def reduce(key, values):
             yield (key, values)
+
+
+class RemoveCollectionRightsTranslatorIdsOneOffJob(
```
Can you please merge these two jobs into one that will remove both fields?

vojtechjelinek reviewed (changes requested):
Thanks! Left a few comments.

iramin committed:
Merge remote-tracking branch 'oppia_origin/develop' into remove-deprecated-collection-models

iramin committed:
fix coverage issues in collection models

vojtechjelinek commented on core/storage/collection/gae_models_test.py:
```diff
@@ -61,6 +61,18 @@ def test_get_collection_count(self):
+        collection_model = collection_models.CollectionModel.get_by_id('id')
+        snapshot_dict = collection_model.compute_snapshot()
+        snapshot_dict['nodes'] = ['node0', 'node1']
+        collection_model = collection_model._reconstitute(snapshot_dict) # pylint: disable=protected-access
+        self.assertNotIn('nodes', collection_model._properties) # pylint: disable=protected-access
+        self.assertNotIn('nodes', collection_model._values) # pylint: disable=protected-access
```
Try to not add any pylint disables if it is not absolutely needed, here you can use `get_version` or you can look how it is handled in `ExplorationRightsModel` (the use of `convert_to_valid_dict`)

vojtechjelinek replied:
Also, should the same test be added for the `CollectionRightsModel`?

vojtechjelinek reviewed (changes requested):
Thanks! Hopefully last two comments.

iramin committed:
Merge remote-tracking branch 'oppia_origin/develop' into remove-deprecated-collection-models

iramin committed:
remove pylint disables in collection models reconstitute test

vojtechjelinek commented:
@iramin Hi, do you plan to continue working on this PR? Do you need help with anything?

iramin commented:
> 
> 
> @iramin Hi, do you plan to continue working on this PR? Do you need help with anything?

@vojtechjelinek I was going to run jobs on my local machine and then submit the job to Sean, but I got [this error](https://github.com/oppia/oppia/wiki/Frequently-Asked-Questions#setup-issues). After fixing it and running locally, I will submit the job.

vojtechjelinek commented:
@iramin Okay, let me know if you need help with anything.

iramin committed:
Merge remote-tracking branch 'origin/develop' into remove-deprecated-collection-models

iramin commented:
> 
> 
> @iramin Okay, let me know if you need help with anything.

@vojtechjelinek In the recent push, I just merged the updates from the develop branch and some of the end to end tests are failing. I took a look and I didn't find anything related to removing deprecated fields from collection models. Do you have any suggestions for this?

vojtechjelinek commented:
@iramin These tests can be flaky sometimes, I have restarted them.

seanlip reviewed (approved):
LGTM for codeowner files, and also confirming that this runs fine. Thanks @iramin!

vojtechjelinek reviewed (approved):
LGTM!

iramin committed:
Merge remote-tracking branch 'origin/develop' into remove-deprecated-collection-models

vojtechjelinek commented:
@aks681 PTAL (I have pinged Akshay through Hangouts too)

aks681 reviewed (approved):
Lgtm as codeowner

iramin committed:
Merge remote-tracking branch 'origin/develop' into remove-deprecated-collection-models
